iothub/service/src/DigitalTwin/Authentication/IotServiceClientCredentials.cs

```diff
@@ -11,7 +12,7 @@
 
 namespace Microsoft.Azure.Devices.Authentication
 {
-    internal abstract class IotServiceClientCredentials : ServiceClientCredentials
+    internal abstract class IotServiceClientCredentials : ServiceClientCredentials, IAuthorizationHeaderProvider
```

Comment:
This is a interface we already have and is generic instead of defining a abstract method called GetSasToken()